### PR TITLE
Dynamic DNS docs

### PR DESCRIPTION
**Pull request by enolfc**
# Summary

Documentation for the Dynamic DNS, taken from https://wiki.egi.eu/wiki/Dynamic_DNS_tutorial

---

**Related issue :**

### CODE REVIEW

**Comment by enolfc** [comment]
@tdviet, please check docs look good

File: content/en/users/cloud-compute/dynamic-dns/_index.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+- NS-update server uses HTTPS protocol, hostname/secret are encrypted as data
+  and not visible during transfer so it is secure to use the update URL
+
+
+- NS-update portal does not store host secret in recoverable form. If you forget
+  the secret of your hostname, simply generate new one via "Show configuration"
+  button in the host edit page. The old secret will be invalid.
+
+
```
Comment (gwarf) [inline content/en/users/cloud-compute/dynamic-dns/_index.md:103]:
```suggestion
```

File: content/en/users/cloud-compute/dynamic-dns/_index.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+- Hostnames/IP addresses are not expired so no need to refresh IP addresses if
+  no changes, it is enough to run once. You can add the following command
+  `curl https://HOSTNAME:SECRET@nsupdate.fedcloud.eu/nic/update` to cloud-init
+  to assign hostname automatically at VM start
+
+- DNS server set Time-to-Live (max time for caching DNS records) to 1 min for
+  dynamic DNS, but MS Windows seems to not respect that. You can clear DNS cache
+  in Windows with `ipconfig /flushdns` command with Administrator account
+{{% /alert %}}
+
```
Comment (gwarf) [inline content/en/users/cloud-compute/dynamic-dns/_index.md:65]:
```suggestion
```

File: content/en/users/cloud-compute/dynamic-dns/_index.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+   curl https://<hostname>:<secret>@nsupdate.fedcloud.eu/nic/update"
+   ```
+
+   where `<hostname>` is the full hostname generated before, e.g.
+   `myserver.fedcloud-tf.fedcloud.eu` and `<secret>` is the secret generated
+   in the previous step.
+
+1. You can also manually edit your registered hostnames in the _Overview_ menu
+   by clicking on the hostname you'd like to manage
+
```
Comment (gwarf) [inline content/en/users/cloud-compute/dynamic-dns/_index.md:53]:
```suggestion
```

File: content/en/users/cloud-compute/dynamic-dns/_index.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+follows:
+
+```plain
+GET /nic/update?hostname=yourhostname&myip=ipaddress
+Host: nsupdate.fedcloud.eu
+Authorization: Basic base64-encoded-auth-string
+User-Agent:
+```
+
+where:
```
Comment (gwarf) [inline content/en/users/cloud-compute/dynamic-dns/_index.md:81]:
```suggestion
where:

```

File: content/en/users/cloud-compute/dynamic-dns/_index.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+   ![Add host](add-host.png)
+
+1. The portal will then show you a secret than can be used for updating
+   the host ip whenever needed. Note it down so you can use it later
+
+1. From the VM you'd like to assign the name to, run a command like follows:
+
+   ```shell
+   curl https://<hostname>:<secret>@nsupdate.fedcloud.eu/nic/update"
```
Comment (gwarf) [inline content/en/users/cloud-compute/dynamic-dns/_index.md:44]:
```suggestion
   curl "https://<hostname>:<secret>@nsupdate.fedcloud.eu/nic/update"
```

File: content/en/users/cloud-compute/dynamic-dns/_index.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  performing the GET request will be used
+
+## Security
+
+- For updating IP address, only hostname and its secret are needed. No user
+  information is stored on VM in any form for updating IP.
+
+- NS-update server uses HTTPS protocol, hostname/secret are encrypted as data
+  and not visible during transfer so it is secure to use the update URL
+
```
Comment (gwarf) [inline content/en/users/cloud-compute/dynamic-dns/_index.md:97]:
```suggestion
```

**Review by gwarf — CHANGES_REQUESTED** [review]
A few minor fixes.
It may be nice to have a hint about using this from `cloud-init`.

A bit unrelated to the doc itself another point would be to discuss the static part of the DNS: `fedcloud-tf.fedcloud.eu` seems a bit long/duplicated.

**Comment by enolfc** [comment]
> A few minor fixes.

incorporated

> It may be nice to have a hint about using this from `cloud-init`.

yup, added it also

> 
> A bit unrelated to the doc itself another point would be to discuss the static part of the DNS: `fedcloud-tf.fedcloud.eu` seems a bit long/duplicated.

Agree. To keep the current schema we could use `vm.fedcloud.eu` instead. It's short enough and does not feel repetitive.

**Review by gwarf — APPROVED** [review]

**Pull request merged**